sceptre/logging.py

```diff
@@ -0,0 +1,20 @@
+from logging import LoggerAdapter, Logger
+from typing import MutableMapping, Any, Tuple
+
+
+class StackLoggerAdapter(LoggerAdapter):
+    def __init__(self, logger: Logger, stack_name: str, extra: dict = None):
+        """A small wrapper around a Logger that prefixes log messages with the stack name.
+
+        :param logger: The logger to wrap
+        :param stack_name: The name of the stack to every log message
+        :param extra: Extra kwargs to add to the log context (if any)
+        """
+        super().__init__(logger, extra or {})
+        self.stack_name = stack_name
```

Comment:
I like this idea. My only question is whether the situation might arise when a StackLoggingAdapter is wrapping another StackLoggingAdapter. If this is a possibility should we detect it in `__init__` and maybe wrap the inner logger instead of the outer one to avoid chaining the stack names?

Reply:
So, _technically_ (so long as you're using the public interface of a logger), wrapping a logger adapter with a logger adapter shouldn't actually break anything. You'd just end up with functionality from Both.

However, if you look at the way that this logger adapter is being used (and the _only_ way it's being used), it's being used on pretty much the first line of the `__init__` method of classes that don't have the logger injected. So in practice, the only way this scenario could happen would be if someone in their custom hook, resolver, or handler did something like: `self.log = MyOtherLoggerAdapter(self.log, self.stack.name)`. Even if they did, it wouldn't really break anything.

Reply:
Sounds good to me.